src/test/java/com/google/jenkins/plugins/computeengine/ConfigAsCodeTest.java

```diff
@@ -0,0 +1,47 @@
+package com.google.jenkins.plugins.computeengine;
+
+import com.cloudbees.plugins.credentials.Credentials;
+import com.cloudbees.plugins.credentials.CredentialsStore;
+import com.cloudbees.plugins.credentials.SystemCredentialsProvider;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import com.google.jenkins.plugins.credentials.oauth.GoogleRobotPrivateKeyCredentials;
+import com.google.jenkins.plugins.credentials.oauth.ServiceAccountConfig;
+import io.jenkins.plugins.casc.misc.ConfiguredWithCode;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import org.apache.commons.io.IOUtils;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class ConfigAsCodeTest {
+
+    @Rule
+    public JenkinsRule r = new JenkinsConfiguredWithCodeRule();
+
+    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
+    public void shouldCreateCloudInstanceFromCode() throws Exception {
+        assertEquals("Zero clouds found", r.jenkins.clouds.size(), 1);
+        ComputeEngineCloud cloud = (ComputeEngineCloud) r.jenkins.clouds.getByName("gce-jenkins-build");
+        assertNotNull("Cloud by name not found", cloud);
+    }
+
+    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
+    public void shouldCreateGCEClientFromCode() throws Exception {
+        String testKey = IOUtils.toString(this.getClass().getResourceAsStream("gce-test-key.json"));
```

Comment:
Any idea why this line returns null on windows?
Anyone can check it on windows?

Reply:
I'm in the process of getting a VM to work for testing, but I think the issue is that the file is not recognized as a resource by the class loader. My first thought would be to move the file into the resources directory under `src/test/resources/com/google/jenkins/plugins/computeengine/`, and that's something I'll test on windows as well.

Reply:
Nevermind on that change, I misread the directory that it was stored under, but everything I've found indicates this is most likely due to how resources are loaded. This question has an interesting potential solution which would be to use getClass().getClassLoader().getResourceAsStream instead: https://stackoverflow.com/questions/2797162/getresourceasstream-is-always-returning-null

Reply:
I have tried it already.. also null :(

Reply:
So I finally got the tests running on a WIndows VM. Each time the test ran it gave this exception: [AccountIdNotSetException](https://github.com/jenkinsci/google-oauth-plugin/blob/develop/src/main/java/com/google/jenkins/plugins/credentials/oauth/GoogleRobotPrivateKeyCredentials.java#L265). However, this was occurring on the last line where `cloud.getClient()` is called.

When I looked at the surefire output, I found that the key was not null and was printed to that output, but then on line 37 where the ServiceAccountConfig is created, the error was "Failed to write json key to file" caused by an IOException when updating the permissions of the new key file that the oauth plugin creates. Failing to set these permissions here means failing to write to the key file used for the credentials which I think explains the AccountIdNotSetException.

I will try to see what else I can do to recreate the null case but it's sounding like there might be a permissions issue rather than a failure to find the file in the path.

Reply:
Ok.. https://github.com/jenkinsci/google-oauth-plugin/issues/13 and https://github.com/jenkinsci/google-oauth-plugin/pull/19

Reply:
Okay, I guess I'm not too surprised that's in the way here. I'm preparing some changes on top of that PR which will make the oauth plugin compatible with JCasC and unblock you here.

Reply:
For now I have skipped running that test on windows..
Added FIXME on test side to fix it in future..